### DISCUSSION TITLE
installing inside ubuntu server 22 fails from both manual install and from VpnHood Access Server

### DISCUSSION THREAD

**Discussion by farzinsb** [vpnhood/VpnHood#180] [Q&A] [answered]
hi, I tried installing vpnhood inside 2 ubuntu 22 server ( VPS ) both didn't work.
here is the os details:
` lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
`

I tried manual installing the dotnet 6 as sdk by microsoft packages.

>  dotnet --info
> .NET SDK (reflecting any global.json):
>  Version:   6.0.111
>  Commit:    b3bb659a9d
> 
> Runtime Environment:
>  OS Name:     ubuntu
>  OS Version:  22.04
>  OS Platform: Linux
>  RID:         ubuntu.22.04-x64
>  Base Path:   /usr/lib/dotnet/sdk/6.0.111/
> 
> global.json file:
>   Not found
> 
> Host:
>   Version:      6.0.11
>   Architecture: x64
>   Commit:       943474ca16
> 
> .NET SDKs installed:
>   6.0.111 [/usr/lib/dotnet/sdk]
> 
> .NET runtimes installed:
>   Microsoft.AspNetCore.App 6.0.11 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.NETCore.App 6.0.11 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]

I changed ssh port back to the default 22 also checked that nothing is active on port443 and gave user pass to the VpnHood Access Server , when it was not a standard ssh port it used to return some errors. but now it doesn't return any errors and yet its not installing it.
then I tried to copy the command from portal to the ssh...  ( 3rd method ) 
it seems to do nothing, no errors after some seconds... still staying on not installed.

so I tried the manual install and adding the restapi code to the config file 
`/opt/VpnHoodServer# vim appsettings.json`

for manual restarting of the vpn server the dll file is missing:

```
/launcher/run.dll
sudo dotnet /opt/VpnHoodServer/launcher/run.dll stop

Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet-/opt/VpnHoodServer/launcher/run.dll does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.

```
can you guide me how should i get that dll file ?

**Comment by trudyhood** [comment, marked as answer]
Hi,
We have improved VpnHood server installation. You don't need to install .net anymore and there is no run.dll.
Please check the following page and let me know if there is a problem.
https://github.com/vpnhood/VpnHood/wiki/How-to-run-VpnHood-Server-on-Linux

**Comment by farzinsb** [comment]
thanks this one is working and I could manually add config file to the json file and connect it to the Access Server panel.
